cypress/support/axe_wrapper.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * Wrapper for axe accessibility testing library.
+ * Provides functionality to enable injection of axe into pages under
+ * test in Cypress, followed by running an a11y audit on the page.
+ */
+
+/**
+ * Inject axe into the page and run a11y audit
+ *
+ * window: browser window instance, typically derived via cy.window()
+ * axeOptions: object containing options to pass directly to axe.run();
+ *     see axe documentation for details
+ */
+function runAxe(window, axeOptions) {
+    axeOptions = axeOptions || {};
+
+    // only inject axe if <script id="axe"> isn't in the page already
+    if (Cypress.$('#axe').length < 1) {
+        const script = window.document.createElement('script');
+        script.id = 'axe';
+        script.async = false;
+        script.innerHTML = require('axe-core/axe.js').source;
+        window.document.body.appendChild(script);
+    }
+
+    // axe adds itself as a property on the window object
+    return window.axe.run(axeOptions).then((results) => {
+        let violations = results.violations;
+        let location = window.location.href.replace(Cypress.config('baseUrl'), '');
+
+        if (violations.length > 0) {
+            let dedupedViolations = dedupeViolations(location, violations);
+
+            // log to the console inside the browser
+            logViolations(dedupedViolations, console.log);
+
+            // this will be logged by cy.task('log', ...)
+            return dedupedViolations;
+        }
+
+        return null;
```

Comment:
Should probably return empty violations list as-is

Reply:
might be cleaner, do we know how the calling code handles this?

Reply:
ignore that - we don't handle the return value.